### PR TITLE
ManagedIdentityCredential: Add support for AppServiceV2019

### PR DESCRIPTION
**Pull request by antkmsft**

### CODE REVIEW

File: sdk/identity/azure-identity/src/managed_identity_source.cpp [hunk 1]
```diff
@@ -35,41 +35,46 @@ Azure::Core::Url ManagedIdentitySource::ParseEndpointUrl(
...
-  auto msiSecret = Environment::GetVariable("MSI_SECRET");
+  auto msiEndpoint = Environment::GetVariable(endpointVarName);
+  auto msiSecret = Environment::GetVariable(secretVarName);
 
   return (msiEndpoint.empty() || msiSecret.empty())
       ? nullptr
-      : std::unique_ptr<ManagedIdentitySource>(new AppServiceManagedIdentitySource(
-          clientId, options, ParseEndpointUrl(msiEndpoint, EndpointVarName), msiSecret));
+      : std::unique_ptr<ManagedIdentitySource>(
+          new T(clientId, options, ParseEndpointUrl(msiEndpoint, endpointVarName), msiSecret));
```
Comment (vhvb1989) [inline sdk/identity/azure-identity/src/managed_identity_source.cpp:51]: I got nervous whenever I read `new` 🐺   .  Consider using `std::make_unique` instead  ( `std::make_unique<T>(arg,arg2,)`)

File: sdk/identity/azure-identity/src/private/managed_identity_source.hpp [hunk 2]
```diff
@@ -28,24 +28,83 @@ namespace Azure { namespace Identity { namespace _detail {
...
+        : AppServiceManagedIdentitySource(
+            clientId,
+            options,
+            endpointUrl,
+            secret,
+            "2017-09-01",
+            "secret",
+            "clientid")
+    {
+    }
```
Comment (vhvb1989) [inline sdk/identity/azure-identity/src/private/managed_identity_source.hpp:76]: should it be better to have this in .cpp ?

**Review by vhvb1989 — APPROVED** [review]

File: sdk/identity/azure-identity/src/managed_identity_source.cpp (hunk 1 above)
Comment (antkmsft) [reply sdk/identity/azure-identity/src/managed_identity_source.cpp:51]: 1. `_detail::ManagedIdentitySource` is not visible to customer. It only has a forward declaration. So the factory methods are not part of public API.

2. `std::make_unique` does require constructor to be public, but so does `new T`. That's why there's "`friend AppServiceManagedIdentitySource`" declaration in `AppServiceV2019ManagedIdentitySource` and in `V2017`. Again, all strictly internal, not visible anywhere, so `friend` is even less of a problem.

Comment (antkmsft) [reply sdk/identity/azure-identity/src/managed_identity_source.cpp:51]: I'll see if I can use `make_unique`.

Comment (antkmsft) [reply sdk/identity/azure-identity/src/managed_identity_source.cpp:51]: Ah yes - then it is the `make_unique` that I'll need to declare a friend. I'd avoid that, that declaration would be cumbersome.

**Pull request merged**